### DISCUSSION TITLE
Programatically add/update/delete users?

### DISCUSSION THREAD

**Discussion by jackie-chan** [apostrophecms/apostrophe#4120] [Q&A] [answered]
Hi there,
Is there a way to programatically add/update/delete users?
Trying to build a backend service that after performing a bunch of different functions first, then manage the user (basic CRUD operations). Is there something that's exposed that allows this?
Thanks!

**Comment by boutell** [comment, marked as answer]
Hi Jackie,

Apostrophe uses its documented REST APIs for everything, including managing users. If you configure an api key with admin privileges you will be able to use that API to manage users. See:

https://v3.docs.apostrophecms.org/reference/api/pieces.html#endpoints

**Reply by boutell** [reply]
Note that the `password` field is a little bit of a special case. If you pass a value for it to `PUT` or `PATCH`, it will update. If you pass an empty string or no password property at all, the existing password is left in place.

**Reply by jackie-chan** [reply]
Thanks so much! I seem to have been in the Guide pages of the documentation rather than the Reference page. Thanks!
